rebasebot/bot.py

```diff
@@ -161,37 +161,31 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
             if not _resolve_rebase_conflicts(gitwd):
                 raise RepoException(f"Git rebase failed: {ex}") from ex
 
-
-def _needs_merge(gitwd, dest):
-    logging.info("Checking if we need a merge commit")
-
+def _prepare_rebase_branch(gitwd, source, dest):
+    logging.info("Preparing rebase branch")
+    
+    # Remove an old merge-tmp branch if it exists
     try:
-        gitwd.git.checkout(f"dest/{dest.branch}")
-        gitwd.git.merge("--no-ff", "rebase")
-    except git.GitCommandError:
-        logging.info("Merge commit is required")
-        return True
-    finally:
-        gitwd.git.reset("--hard", "HEAD")
-        gitwd.git.checkout("rebase")
+        gitwd.git.branch("-d", "merge-tmp", force=True)
```

Comment:
Nit: i thinking about constant for this "merge-tmp". Not necessary, but would look a bit better, imho